### ISSUE TITLE
CEF3: target=_blank crashes the browser with CustomScheme

### ISSUE THREAD

**Issue by neopobo** [cefsharp/CefSharp#910]
I am trying to implement CustomeScheme exactly as it is described in:
http://thechriskent.com/2014/04/21/use-local-files-in-cefsharp/
git repo
[https://github.com/Dirkster99/KB/tree/2797a540772efb45912488279d7fa73c64ccf58d/00_HelloWorld/KnowledgeBase%20-%20Sample%204%20LocalSchemeHandler]

I have notice that, when there is a link inside the local html file (eg. local://web/index.html) with target="_blank" attribute, then browser process crashes.
It doesn't mater if the links points local address like:
`<a href="local://web/home.html" target="_blank">local _blank</a>`
or http address:
`<a href="http://google.com" target="_blank">google _blank</a>`

It happens only with links from local file (loaded with local:\\ scheme).
When target = "_blank" is on the link from the web, then it works and new window is open.
So it works with test url from #198 issue.

I have try with Nuget packages:
id="CefSharp.Wpf" version="37.0.0" targetFramework="net45" 
and
id="CefSharp.Wpf" version="39.0.0-pre03" targetFramework="net45" 

I have also tried to implement OnBeforePopup from ILifeSpanHandler, but unfortunately this one not fires at all. The crash have to happen before it.
Only OnRenderProcessTerminated from IRequestHandler fires, but of-course with status "ProcessCrashed".

Anybody has an idea what can be the reason of such problems?

**Comment by amaitland** [comment]
Looks like a `CEF` problem, if you search for `_blank` you'll see there are a number of open issues.

https://bitbucket.org/chromiumembedded/cef/issues?q=_blank

(`CefSharp` being a wrapper around `CEF`, problems need to be fixed in `CEF`).

**Comment by amaitland** [comment]
Can you set Logging to Verbose and post that (if it's large, best to post as a link, maybe as a `GIST`).

I would say that it's related to this issue
https://bitbucket.org/chromiumembedded/cef/issue/1000/cef3-crash-when-opening-popup-window-in

If you do decide to post something on the upstream, the following will be useful:
`CefSharp.WPF` uses `OSR` mode, and version `39` is based on `2171`.

**Comment by neopobo** [comment]
Here comes logs:

[0328/101828:VERBOSE1:pref_proxy_config_tracker_impl.cc(148)] 000000D04543A780: set chrome proxy config service to 000000D04543A000
[0328/101828:VERBOSE1:pref_proxy_config_tracker_impl.cc(277)] 000000D04543A780: Done pushing proxy to UpdateProxyConfig
[0328/101829:VERBOSE1:child_thread.cc(239)] Mojo is disabled on child
[0328/101829:VERBOSE1:webrtc_internals.cc(56)] Could not get the download directory.
[0328/101829:VERBOSE1:child_thread.cc(239)] Mojo is disabled on child
[0328/101829:ERROR:renderer_main.cc(207)] Running without renderer sandbox
[0328/101830:VERBOSE1:texture_image_transport_surface.cc(381)] Allocating new backbuffer texture
[0328/101830:VERBOSE1:gles2_cmd_decoder.cc(2526)] GL_OES_packed_depth_stencil supported.
[0328/101830:VERBOSE1:cookie_monster.cc(2115)] WARNING: Unsupported cookie scheme: local
[0328/101830:VERBOSE1:resource_loader.cc(318)] OnResponseStarted: local://web/index.html
[0328/101830:VERBOSE1:resource_loader.cc(358)] OnReadCompleted: "local://web/index.html" bytes_read = 1074
[0328/101830:VERBOSE1:resource_loader.cc(358)] OnReadCompleted: "local://web/index.html" bytes_read = 0
[0328/101830:VERBOSE1:resource_loader.cc(647)] ResponseCompleted: local://web/index.html
[0328/101830:VERBOSE1:cookie_monster.cc(2115)] WARNING: Unsupported cookie scheme: local
[0328/101830:VERBOSE1:resource_loader.cc(318)] OnResponseStarted: local://web/images/truck.png
[0328/101830:VERBOSE1:resource_loader.cc(358)] OnReadCompleted: "local://web/images/truck.png" bytes_read = 2629
[0328/101830:VERBOSE1:resource_loader.cc(358)] OnReadCompleted: "local://web/images/truck.png" bytes_read = 0
[0328/101830:VERBOSE1:resource_loader.cc(647)] ResponseCompleted: local://web/images/truck.png
[0328/101832:VERBOSE1:ipc_sync_channel.cc(316)] Received error reply
[0328/101832:VERBOSE1:ipc_sync_channel.cc(316)] Received error reply
[0328/101832:ERROR:render_process_host_impl.cc(1429)] bad message 132032 terminating renderer.
[0328/101832:VERBOSE1:ipc_sync_channel.cc(386)] Canceling pending sends
[0328/101832:WARNING:message_in_transit_queue.cc(18)] Destroying nonempty message queue
[0328/101832:VERBOSE1:ipc_sync_channel.cc(386)] Canceling pending sends
[0328/101832:VERBOSE1:ipc_sync_channel.cc(386)] Canceling pending sends

**Comment by amaitland** [comment]
(Best to format such big blocks, when you post there's `Markdown Supported` which will guide you).

```
[0328/101832:ERROR:render_process_host_impl.cc(1429)] bad message 132032 terminating renderer
```

I'd say that's a problem with `CEF`. You can try the https://github.com/cefsharp/CefSharp/tree/cef/2272 branch of `CefSharp` to see if a newer version resolves it. Probably best to report `upstream`.

**Comment by amaitland** [comment]
(I edited your comment containing the log and formatted it).

**Comment by neopobo** [comment]
Thank You amaitland for your analysis.
I am going to try with cef/2272 branch.

**Comment by neopobo** [comment]
I have tried with cef/2272 branch.
I it also not working here.

I have notice, that there is already made example with target = “_blank” attribute.
Basically inside CefSharp.Wpf.Example app, we can put 
```
custom://cefsharp/PopupTest.html url, 
```
and then anybody can test this problem.

My logs from WPF example app, are similar:
```
[0330/151625:VERBOSE1:pref_proxy_config_tracker_impl.cc(148)] 08585538: set chrome proxy config service to 04A433A8
[0330/151625:VERBOSE1:pref_proxy_config_tracker_impl.cc(277)] 08585538: Done pushing proxy to UpdateProxyConfig
[0330/151625:VERBOSE1:child_thread.cc(244)] Mojo is disabled on child
[0330/151626:VERBOSE1:webrtc_internals.cc(55)] Could not get the download directory.
[0330/151626:VERBOSE1:child_thread.cc(244)] Mojo is disabled on child
[0330/151626:ERROR:renderer_main.cc(212)] Running without renderer sandbox
[0330/151626:VERBOSE1:gles2_cmd_decoder.cc(2593)] GL_OES_packed_depth_stencil supported.
[0330/151626:VERBOSE1:cookie_monster.cc(2130)] WARNING: Unsupported cookie scheme: custom
[0330/151626:VERBOSE1:resource_loader.cc(364)] OnResponseStarted: custom://cefsharp/BindingTest.html
[0330/151626:VERBOSE1:resource_loader.cc(434)] OnReadCompleted: "custom://cefsharp/BindingTest.html" bytes_read = 4275
[0330/151626:VERBOSE1:resource_loader.cc(434)] OnReadCompleted: "custom://cefsharp/BindingTest.html" bytes_read = 0
[0330/151626:VERBOSE1:resource_loader.cc(778)] ResponseCompleted: custom://cefsharp/BindingTest.html
[0330/151640:VERBOSE1:cookie_monster.cc(2130)] WARNING: Unsupported cookie scheme: custom
[0330/151640:VERBOSE1:resource_loader.cc(364)] OnResponseStarted: custom://cefsharp/PopupTest.html
[0330/151640:VERBOSE1:resource_loader.cc(434)] OnReadCompleted: "custom://cefsharp/PopupTest.html" bytes_read = 340
[0330/151640:VERBOSE1:resource_loader.cc(434)] OnReadCompleted: "custom://cefsharp/PopupTest.html" bytes_read = 0
[0330/151640:VERBOSE1:resource_loader.cc(778)] ResponseCompleted: custom://cefsharp/PopupTest.html
[0330/151656:ERROR:render_process_host_impl.cc(1539)] bad message 132082 terminating renderer.
[0330/151656:VERBOSE1:ipc_sync_channel.cc(386)] Canceling pending sends
[0330/151656:VERBOSE1:ipc_sync_channel.cc(386)] Canceling pending sends
[0330/151656:VERBOSE1:ipc_sync_channel.cc(386)] Canceling pending sends 
```

**Comment by amaitland** [comment]
Closing in favor of #1263
